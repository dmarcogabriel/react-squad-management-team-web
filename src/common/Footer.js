import React from 'react';

const Footer = () => (
  <div className="bg-footer py-3 flex justify-center">
    <p className="text-sm text-gray-900">2020 - All rights reserved</p>
  </div>
);

export default Footer;
