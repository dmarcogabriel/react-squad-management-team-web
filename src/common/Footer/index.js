import React from 'react';

const Footer = () => (
  <div data-testid="footer" className="bg-footer py-3 flex justify-center">
    <p data-testid="footer-text" className="text-sm text-gray-900">
      2020 - All rights reserved
    </p>
  </div>
);

export default Footer;
