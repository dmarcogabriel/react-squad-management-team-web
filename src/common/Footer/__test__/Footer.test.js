import { render, cleanup } from '@testing-library/react';
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Footer from '..';

afterEach(cleanup);

it('renders Footer without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Footer />, div);
});

it('renders Footer classes correctly', () => {
  const { getByTestId } = render(<Footer />);

  expect(getByTestId('footer')).toHaveClass(
    'bg-footer py-3 flex justify-center'
  );
});

it('renders Footer text correctly', () => {
  const { getByTestId } = render(<Footer />);

  expect(getByTestId('footer-text')).toHaveTextContent(
    '2020 - All rights reserved'
  );
});

it('matches snapshot', () => {
  const tree = renderer.create(<Footer />).toJSON();
  expect(tree).toMatchSnapshot();
});
