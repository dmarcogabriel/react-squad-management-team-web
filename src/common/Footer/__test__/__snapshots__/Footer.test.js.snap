// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="bg-footer py-3 flex justify-center"
  data-testid="footer"
>
  <p
    className="text-sm text-gray-900"
    data-testid="footer-text"
  >
    2020 - All rights reserved
  </p>
</div>
`;
